Divide eav attribute cache per website

vshcheglov opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

## Actual pull request -> https://github.com/magento/magento2/pull/34856

### Description (*)
We have an ability to choose different attribute settings per website in Magento admin, but have no per website eav attribute cache separation. So our attribute settings will be depended on which website will be opened first after eav cache cleaning.

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. Fixes magento/magento2#10978

### Manual testing scenarios (*)
1. Open clean Magento instance

Creating new website and store
1. Go to Stores -> Settings -> All Stores
1. Select Create Website
1. Create a new website
1. Select Create Store
1. Create a new store and assign it to a newly created website
1. Select Create Store View
1. Create a new enabled store view and assign it to newly created store
1. Remember store view code
1. Go to System -> Cache Management
1. Flush Cache Storage

Filling required settings
1. Go to Stores -> Configuration -> Customers -> Customer Configuration -> Name and Address Options
1. Select Main Website scope
1. Set Date of Birth to No
1. Set Show Tax/VAT Number to Optional
1. Save config
1. Select new website scope
1. Set Date of Birth to Optional
1. Set Show Tax/VAT Number to No
1. Save config

Enabling store code in urls for fast website switching
1. Select Default Config scope
1. Go to General -> Web -> Url Options
1. Change Add Store Code to Urls to Yes
1. Save config

Reproducing the bug
1. Go to System -> Cache Management
1. Flush cache storage
1. Important! First of all, go to Main Website Storefront Create Account Page
1. Go to https://yourmagento.com/default/customer/account/create/
1. We see Tax/Vat Number field and don't see Dob field
1. Then go to new created website Create Account Page
1. Go to https://yourmagento.com/new-store-view-code/customer/account/create/

Expected:
1. We don't see Tax/Vat Number field and see Dob field

Actual:
1. We see Tax/Vat Number field and don't see Dob field

Then another case:
1. Flush cache storage
1. Important! First of all, go to New Website Storefront Create Account Page
1. Go to yourmagento.com/new-store-view-code/customer/account/create/
1. We don't see Tax/Vat Number field and see Dob field (as expected sooner above)
1. Now go to Main Website Storefront create account page
1. Go to https://yourmagento.com/default/customer/account/create/

Expected:
1. We see Tax/Vat Number field and don't see Dob field

Actual:
1. We don't see Tax/Vat Number field and see Dob field


### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

vshcheglov commented:
@magento give me 2.5-develop instance

rikwillems commented:
I just added a suggestion to the PR.

rikwillems commented:
@vshcheglov please see my two PRs to yours. The current implementation is not working for me without these changes.

vshcheglov commented:
> @vshcheglov please see my two PRs to yours. The current implementation is not working for me without these changes.

Yeah, I missed some changes. Thank you!

danielrussob commented:
@magento give me 2.5-develop instance

danielrussob commented:
@magento give me test instance

aleron75 commented:
@magento run all tests

danielrussob commented:
I have reproduced the issue on vanilla magento 2.5 instance

aleron75 commented:
Hello @vshcheglov!

Thanks for your contribution.

Due to [Magento Definition of Done](https://devdocs.magento.com/contributor-guide/contributing_dod.html#automated-tests), all code must be covered by tests.

For this specific case, this fix should be covered by automated tests with the scenario which leads to an issue: tests should fail on the mainline and pass with this fix.

A good candidate would be extending the `\Magento\Eav\Test\Unit\Model\ConfigTest` unit test.

Thank you!

danielrussob commented:
@magento give me test instance

aleron75 commented:
One last thing @vshcheglov: generally only features and tasks with BIC are allowed to be introduced into the `2.5-develop` branch. 

All the bugfix should be delivered to the `2.4-develop` branch;  all fixes delivered to `2.4-develop` will be automatically ported to `2.5-develop`.

So, please, submit this bugfix to the `2.4-develop` branch.

Thanks.

Nazar65 commented:
Hi @vshcheglov  do we need to do the same with child's products for configurable, it also uses same cache for front and back, and also store here : -> https://github.com/magento/magento2/blob/0a05fa3da4c645926af0e6d8dea1c81d819be8fc/app/code/Magento/ConfigurableProduct/Model/Product/Type/Configurable.php#L1436

vshcheglov commented:
@aleron75 
> So, please, submit this bugfix to the 2.4-develop branch.

patch-2 branch is based on 2.5-develop branch so we should not merge it to 2.4

I prepared a new branch based on 2.4.develop branch but don't see any ability to change patch-2 branch to a new branch.
How to deal with it? I can create a new pull request with a new branch.

@Nazar65 
> Hi @vshcheglov do we need to do the same with child's products for configurable, it also uses same cache for front and back, and also store here : ->

I am not sure that it is related to the current issue.

sidolov commented on app/code/Magento/Eav/Model/Config.php:
```diff
@@ -960,4 +970,14 @@ private function initAttributesFromCache(Type $entityType)
+     * 
+     * @return int
+     */
+    private function getWebsiteId()
```
Please, add return type

Rrego6 committed:
AC-1453: Fix Static/Unit test

- Fix static tests
- Fix unit tests

Commit:
fix for static tests

karyna-t committed:
fixes for unit tests php 8.1

adifucan committed:
MCP-771: [MCP-608] Multiple consumers prototype development

Commit:
fix after a CR

slopukhov committed:
Merge remote-tracking branch 'performance/MCP-773' into MCP-773

Rrego6 committed:
Merge remote-tracking branch 'mainline/2.4-develop' into AC-1453-test2

Rrego6 committed:
Merge remote-tracking branch 'mainline/2.4-develop' into AC-1453-test2

Rrego6 committed:
Merge remote-tracking branch 'mainline/2.4-develop' into AC-106

Rrego6 committed:
Merge remote-tracking branch 'trigger/AC-106' into AC-1453-test

slopukhov committed:
MCP-773: [GraphQL] Error in applying a coupon during a checkout

fascinosum committed:
Merge pull request #337 from magento-trigger/AC-1443

AC-1443: Unable to locate Row Content Type in PageBuilder

Rrego6 committed:
AC-1453: Fix Static/Unit test

- Fix static tests

avattam06 committed:
Merge branch '2.4-develop' into B2B-2060

oshmyheliuk committed:
Merge remote-tracking branch 'origin/2.4-develop' into B2B-2048

avattam06 committed:
B2B-2067: Gallery Image is not saving the remote storage file on to local

- Added fix for the test on Gallery class

Rrego6 committed:
Merge remote-tracking branch 'mainline/2.4-develop' into AC-117

Rrego6 committed:
Merge pull request #341 from magento-trigger/AC-1453-test

AC-1453: Fix Static/Unit test

Rrego6 committed:
Merge pull request #339 from magento-trigger/AC-117

AC-117: Synchronous XMLHttpRequest warning message

kieuphan2406 committed:
Merge remote-tracking branch 'trigger/AC-1437' into AC-106

kieuphan2406 committed:
Merge remote-tracking branch 'trigger/AC-1567' into AC-106

magento-engcom-team committed:
ENGCOM-9317: Replace repetitive actions with Action Groups in AdminCreateDuplicateCategoryTest and AdminCreateDuplicateProductTest #34410

 - Merge Pull Request magento/magento2#34410 from kate-kyzyma/magento2:Refactoring-AdminCreateDuplicateCategoryTest
 - Merged commits:
   1. 9972de05b32557998674a528c7b32c3f82dde0b5
   2. c934b510419356b9e369c185bb3b69d055e18897
   3. ef654e5b195d9bb46bbd89b479f3cb17173cc0fc
   4. 1faca0100ff04513e1c306454c02d703890d7af1

magento-engcom-team committed:
ENGCOM-9254: Resolve height modal issue on iOS #34290

 - Merge Pull Request magento/magento2#34290 from mrtuvn/magento2:resolve-height-modal-scroll-inner-on-mobile
 - Merged commits:
   1. ea1ab67b6992b2fabd1f7390747487a7b307f8fa

magento-engcom-team committed:
ENGCOM-9188: Fix accessibility errors on cart and checkout pages #33851

 - Merge Pull Request magento/magento2#33851 from vovayatsyuk/magento2:fix-accessibility-on-cart-and-checkout
 - Merged commits:
   1. 6c1807be1904232b877dabaadff8e6a810d16b30
   2. a35ee4e05f591c9ab9cf28326ffdaa55c541a01f
   3. a9e5f7b32acb19856b57cada7afbcb2ea9b7c4e5
   4. 8c0e20ca32d067e20f7162c53b94f8a57159ebf4
   5. 9f676cb39ed99b57d683c150c3e896549385eb26
   6. 777e337c5daa58b256fe69bf563d52b39dfda3e3
   7. e0197b9bb9296196db80e1432257a9d1a31e3259

oshmyheliuk committed:
B2B-2024: [AWS S3] [Integration Tests]: Investigate Test Failures in cms module

kate-kyzyma committed:
Refactoring-AdminUpdateFlatCategoryAndAddProductsTest

kate-kyzyma committed:
Merge branch '2.4-develop' into Refactoring-AdminCreateInactiveFlatCategoryTest

kate-kyzyma committed:
Merge branch 'Refactoring-AdminUpdateFlatCategoryAndAddProductsTest' into Refactoring-AdminCreateInactiveFlatCategoryTest

kate-kyzyma committed:
Refactoring-AdminUpdateFlatCategoryNameAndDescriptionTest

avattam06 committed:
Merge remote-tracking branch 'origin/B2B-2060' into B2B-2022

slopukhov committed:
MCP-770: Add performance measurement of checkout with a large number of products in the cart for regular PAT build

NadiyaS committed:
B2B-2049: [AWS S3] Magento\Customer\Model\Metadata\Form\Image File Filesystem hardcoded

ArtemVasyliev committed:
Merge branch 'ACP2E-213' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

ArtemVasyliev committed:
Merge branch 'MC-42750' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

ArtemVasyliev committed:
Merge branch 'ACP2E-51' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

ArtemVasyliev committed:
Merge branch 'ACP2E-7' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

ArtemVasyliev committed:
Merge branch 'ACP2E-212' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

ArtemVasyliev committed:
Merge branch 'ACP2E-216' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

ArtemVasyliev committed:
Merge branch 'MC-43163' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

ArtemVasyliev committed:
Merge branch 'ACP2E-31' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

ArtemVasyliev committed:
Merge branch 'ACP2E-82' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

ArtemVasyliev committed:
Merge branch 'ACP2E-147' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

ArtemVasyliev committed:
Merge branch 'ACP2E-227' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

ArtemVasyliev committed:
Merge branch 'ACP2E-107' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

ArtemVasyliev committed:
Merge branch 'ACP2E-169' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

avattam06 committed:
Merge branch '2.4-develop' into B2B-2028

NadiyaS committed:
B2B-2049: [AWS S3] Magento\Customer\Model\Metadata\Form\Image File Filesystem hardcoded

avattam06 committed:
Merge branch 'magento-commerce:2.4-develop' into B2B-2058

avattam06 committed:
Merge remote-tracking branch 'origin/B2B-2058' into B2B-2022

danmooney2 committed:
B2B-1986: Get Cloud-Native-Testing Integration Test Builds to Work Against Aurora

oshmyheliuk committed:
B2B-2024: [AWS S3] [Integration Tests]: Investigate Test Failures in cms module

vzabaznov committed:
MCP-771: [MCP-608] Multiple consumers prototype development

vzabaznov committed:
MCP-771: [MCP-608] Multiple consumers prototype development

vzabaznov committed:
MCP-771 [MCP-608] Multiple consumers prototype development

vzabaznov committed:
MCP-771 [MCP-608] Multiple consumers prototype development

oshmyheliuk committed:
B2B-2024: [AWS S3] [Integration Tests]: Investigate Test Failures in cms module

sivaschenko committed:
AC-156: Updated Magento Coding Standard to version 14

glo05363 committed:
Merge remote-tracking branch 'origin' into ac1449
merge master

avattam06 committed:
B2B-2061: Merge

avattam06 committed:
Merge remote-tracking branch 'origin/B2B-2048' into B2B-2022

avattam06 committed:
Merge remote-tracking branch 'origin/B2B-2067' into B2B-2022

avattam06 committed:
Merge branch 'B2B-2067-branch' into HEAD

avattam06 committed:
Merge branch 'B2B-2022' of github.com:magento-arcticfoxes/magento2ce into B2B-2022

avattam06 committed:
Merge remote-tracking branch 'origin/B2B-1986' into B2B-2022

avattam06 committed:
B2B-2022:[AWS S3] [Integration Tests]: Investigate Test Failures in Catalog module

- removing unwanted files on PR

kieuphan2406 committed:
AC-106: Upgrade jQuery UI to latest version 1.12.1
Fixed spinner in captcha test

adifucan committed:
MCP-771: [MCP-608] Multiple consumers prototype development

adifucan committed:
MCP-771: [MCP-608] Multiple consumers prototype development

adifucan committed:
MCP-771: [MCP-608] Multiple consumers prototype development

- Fix Unknown connection name amqp error;

adifucan committed:
MCP-771: [MCP-608] Multiple consumers prototype development

- Fix Unknown connection name amqp error;

sidolov committed:
Merge pull request #7158 from magento-gl/ac1449

Not able to complete simple/configurable product creation test flow a…

fascinosum committed:
Merge branch 'magento-commerce:2.4-develop' into AC-106

NadiyaS committed:
B2B-2049: [AWS S3] Magento\Customer\Model\Metadata\Form\Image File Filesystem hardcoded

NadiyaS committed:
Merge branch '2.4-develop' of github.com:magento-arcticfoxes/magento2ce into B2B-2049

NadiyaS committed:
B2B-2049: [AWS S3] Magento\Customer\Model\Metadata\Form\Image File Filesystem hardcoded

magento-engcom-team committed:
ENGCOM-9320: Replace repetitive actions with Action Groups in AdminCreateInactiveInMenuFlatCategoryTest #34418

 - Merge Pull Request magento/magento2#34418 from kate-kyzyma/magento2:Refactoring-AdminCreateInactiveInMenuFlatCategoryTest
 - Merged commits:
   1. 94a7cdc71175841cabea52d597fe6d45edea716f

Commit:
Apply suggestions from code review

Commit:
Merge branch '2.4-develop' into Refactoring-CreateProductAttributeEntityWithReservedKeysTest

magento-engcom-team committed:
ENGCOM-9310: Guard Magento_Catalog JavaScript from lack of minicart #34185

 - Merge Pull Request magento/magento2#34185 from fredden/magento2:checkout-no-minicart
 - Merged commits:
   1. 209b24bcb9e449123689817512e65d8e9eda32e0
   2. bdfd02ae0d06943a9aa440f3da6e489019ca77b0

magento-engcom-team committed:
ENGCOM-9321: Replace repetitive actions with Action Groups in AdminConfigurableProductAddConfigurationTest, AdminConfigurableProductDisableAnOptionTest, AdminConfigurableProductRemoveAnOptionTest, AdminConfigurableProductRemoveConfigurationTest #34400

 - Merge Pull Request magento/magento2#34400 from kate-kyzyma/magento2:Refactoring-AdminConfigurableProductAddConfigurationTest
 - Merged commits:
   1. 2d2a20a4fd7bae5adb1095248a71d3b33acd068a
   2. 3f45e23a43d807fadbdccc53f9d59dbaf3fda394

xmav committed:
Merge pull request #7185 from magento-trigger/AC-106

[Trigger] jQuery-UI upgrade

magento-engcom-team committed:
ENGCOM-9322: Replace repetitive actions with Action Groups in AdminAddInStockProductToTheCartTest  #34298

 - Merge Pull Request magento/magento2#34298 from kate-kyzyma/magento2:Refactoring-AdminAddInStockProductToTheCartTest
 - Merged commits:
   1. 90e3b3a555e4a78b6a001850de9e5a009d47d145
   2. 8710edf8487cc31bf72efe63253f6eef2dc02229
   3. a3479ea072924e6d9a3c6596e6d5deb1cc0d9ca1
   4. a1290c6e0c67695920ac3ee1ff3be86bee0b16c0
   5. f548447222dc01519d351ad01af32e63bc6fe657

kate-kyzyma committed:
Fixing Semantic Version Checker issue

kate-kyzyma committed:
Merge branch '2.4-develop' into Refactoring-AdminCreateInactiveFlatCategoryTest

sivaschenko committed:
Merge branch '2.4-develop' of github.com:magento-commerce/magento2ce into magento-coding-standard-v14

magento-engcom-team committed:
ENGCOM-9282: Replace repetitive actions with Action Groups in CreateProductAttributeEntityWithReservedKeysTest #34419

 - Merge Pull Request magento/magento2#34419 from kate-kyzyma/magento2:Refactoring-CreateProductAttributeEntityWithReservedKeysTest
 - Merged commits:
   1. ec0e64904fe0b4795fa3059f89f45c867870d3a5
   2. 2377b2fc4f860a1f9f6c7f0bd0905d9c1ccf679c
   3. 49796bb78d69061ff07a13a235b9813c37e291e9

magento-engcom-team committed:
ENGCOM-9323: Replace repetitive actions with Action Groups in AdminSimpleProductSetEditContentTest,AdminCreateAndEditSimpleProductSettingsTest, AdminCreateAndEditVirtualProductSettingsTest and AdminConfigurableProductBulkUpdateTest #34302

 - Merge Pull Request magento/magento2#34302 from kate-kyzyma/magento2:Refactoring-AdminSimpleProductSetEditContentTest
 - Merged commits:
   1. 67c83d9f2b6d4ea44ab8e38711adb92d53b4e6c6
   2. f550f195db5b616eac28d6f4d3f6ff1d01bc6898
   3. 48162ef55dee01da99c998d582e57446cf80e621
   4. 9a7145222a011326feb5de172673fa0b69ea0e3e

oshmyheliuk committed:
B2B-2024: [AWS S3] [Integration Tests]: Investigate Test Failures in cms module

oshmyheliuk committed:
B2B-2024: [AWS S3] [Integration Tests]: Investigate Test Failures in cms module

sidolov committed:
Merge branch '2.4-develop' into 2.4-develop-fast-lane-prs

NadiyaS committed:
B2B-2049: [AWS S3] Magento\Customer\Model\Metadata\Form\Image File Filesystem hardcoded

sivaschenko committed:
Merge pull request #7191 from magento-lynx/magento-coding-standard-v14

[Lynx] Updated Magento Coding Standard to version 14

avattam06 committed:
Merge branch 'magento-commerce:2.4-develop' into B2B-2022

avattam06 committed:
Merge branch '2.4-develop' into B2B-2028

slopukhov committed:
Merge branch '2.4-develop' of github.com:magento-commerce/magento2ce into MCP-770

slopukhov committed:
Merge branch '2.4-develop' of github.com:magento-commerce/magento2ce into MCP-773

adifucan committed:
Merge remote-tracking branch 'origin/2.4-develop' into MCP-771

oshmyheliuk committed:
Merge branch '2.4-develop' into B2B-2024

oshmyheliuk committed:
B2B-2024: [AWS S3] [Integration Tests]: Investigate Test Failures in cms module

adifucan committed:
MCP-771: [MCP-608] Multiple consumers prototype development

- Set amqp connection for WebapiAsync;

adifucan committed:
MCP-771: [MCP-608] Multiple consumers prototype development

- Set amqp connection for WebapiAsync;

avattam06 committed:
B2B-2022:[AWS S3] [Integration Tests]: Investigate Test Failures in Catalog module

- Add static fix failures

avattam06 committed:
Merge branch 'B2B-2022' of github.com:magento-arcticfoxes/magento2ce into B2B-2022

cpartica committed:
Merge remote-tracking branch 'mainline/2.4-develop' into graphql-doc-changes

# Conflicts:
#	app/code/Magento/CatalogGraphQl/etc/schema.graphqls
#	app/code/Magento/QuoteGraphQl/etc/schema.graphqls

amolina-adobe committed:
DOC-474: Fixing failed static test

Commit:
magento-engcom/magento2ce#4266: Fixed MFTF Test failure

sidolov committed:
Merge branch '2.4-develop' into 2.4-develop-fast-lane-prs

adifucan committed:
MCP-771: [MCP-608] Multiple consumers prototype development

- Update unit tests;

adifucan committed:
MCP-771: [MCP-608] Multiple consumers prototype development

- Update unit tests;

adifucan committed:
MCP-771: [MCP-608] Multiple consumers prototype development

- Fix static tests;

adifucan committed:
MCP-771: [MCP-608] Multiple consumers prototype development

- Fix static tests;

NadiyaS committed:
B2B-2049: [AWS S3] Magento\Customer\Model\Metadata\Form\Image File Filesystem hardcoded

NadiyaS committed:
Merge branch '2.4-develop' of github.com:magento-arcticfoxes/magento2ce into B2B-2049

victor-v-rad committed:
Merge branch '2.4-develop' into ACP2E-10

# Conflicts:
#	app/code/Magento/SalesRule/Model/RulesApplier.php

victor-v-rad committed:
ACP2E-10: Incorrect Discount: Two Cart rules with and without coupon

avattam06 committed:
Merge pull request #7161 from magento-arcticfoxes/B2B-2022

B2B-2022 :[AWS S3] [Integration Tests]: Investigate Test Failures in Catalog module

avattam06 committed:
Merge branch 'magento-commerce:2.4-develop' into B2B-2028

Commit:
Merge remote-tracking branch 'MJTheOne/2.4-develop' into 2.4-develop-fast-lane-prs

sidolov committed:
Merge branch '2.4-develop' into 2.4-develop-fast-lane-prs

victor-v-rad committed:
ACP2E-10: Incorrect Discount: Two Cart rules with and without coupon

sidolov committed:
:arrows_clockwise: [Magento Community Engineering] Community Contributions - 2.4-develop latest changes

Accepted Community Pull Requests:
 - magento/magento2#34399: Speedup DataObject->addData when used on an empty object (by @thlassche)
 - magento/magento2#34346: Replace repetitive actions with Action Groups in AdminSimpleSetEditRelatedProductsTest (by @kate-kyzyma)
 - magento/magento2#32792: [MFTF] add StorefrontGiftMessageForOrderOnCheckoutCartPageTest  (by @Usik2203)
 - magento/magento2#33454: [MFTF] Check if Shipping Address added on Checkout is saved after Order is placed (by @AnnaAPak)
 - magento/magento2#34022: [MFTF] StorefrontCheckRefundGrandTotalActionGroup adding (by @AnnaAPak)
 - magento/magento2#33999: [MFTF] AdminAddNewDefaultBillingShippingCustomerAddressTest refactoring (by @AnnaAPak)
 - magento/magento2#34256: Replace repetitive actions with Action Groups in StorefrontProductNameWithDoubleQuoteTest (by @kate-kyzyma)
 - magento/magento2#34110: Replace repetitive actions with Action Groups in AdminNavigateMultipleUpSellProductsTest (by @kate-kyzyma)
 - magento/magento2#34099: Replace repetitive actions with Action Groups in AdminEditRelatedBundleProductTest, EnableDisableBundleProductStatusTest, AdminMassDeleteBundleProductsTest (by @kate-kyzyma)
 - magento/magento2#34257: Replace repetitive actions with Action Groups in AdminConfigurableProductBulkDeleteTest (by @kate-kyzyma)
 - magento/magento2#34051: Replace repetitive actions with Action Groups in StorefrontConfigurableProductGridViewTest and StorefrontConfigurableProductListViewTest (by @kate-kyzyma)
 - magento/magento2#34076: Replace repetitive actions with Action Groups in AdminProductBundleCreationTest (by @kate-kyzyma)
 - magento/magento2#34075: Replace repetitive actions with Action Groups in AdminAddBundleItemsTest (by @kate-kyzyma)
 - magento/magento2#33530: Removed CliCacheFlushActionGroup usage for CatalogSearch and other modules (by @sergiy-v)


Fixed GitHub Issues:
 - magento/magento2#32821: [Issue] [MFTF] add StorefrontGiftMessageForOrderOnCheckoutCartPageTest  (reported by @m2-assistant[bot]) has been fixed in magento/magento2#32792 by @Usik2203 in 2.4-develop branch
   Related commits:
     1. 716ee723402491a503f07c5d9390d83c279ab3b6
     2. 3f0219e606af27c47226ccd15981ac2cc62a8793
     3. 90e9b12060fe3592db6ee677d6b1a18aa8e088f2
     4. a1a91ae097ffe8950a2fdd706a120368c7c3607b
     5. d7a2d85d0004a2dabd3c539f7a0f39d58b0e0e5c
     6. c6016f53dd51e6c618131e2c43409f93f27eedd9
     7. 7502723946b97eb3234230a574f63186e23812b4
     8. ab39c95ac30bdd30993d82462330dacee4b9b7a9
     9. 07a6b7781a56a11ec146240243bc95dc07258d97

 - magento/magento2#34315: [Issue] [MFTF] StorefrontCheckRefundGrandTotalActionGroup adding (reported by @m2-assistant[bot]) has been fixed in magento/magento2#34022 by @AnnaAPak in 2.4-develop branch
   Related commits:
     1. 865bf66c8955f18d86052bc8d9d0e6e517b4f25a
     2. 0919deb92ceff2d7cebfa0d58ac429986b12d09e
     3. 9eba2c18260f9afb53cd49670bd1abab8e726346

 - magento/magento2#34317: [Issue] Replace repetitive actions with Action Groups in StorefrontProductNameWithDoubleQuoteTest (reported by @m2-assistant[bot]) has been fixed in magento/magento2#34256 by @kate-kyzyma in 2.4-develop branch
   Related commits:
     1. 72d76fcd3005722bda9f167f163363ee486c3947
     2. aafadcfc6549a043ba9208cce097c5e652a613be

 - magento/magento2#34314: [Issue] Replace repetitive actions with Action Groups in AdminNavigateMultipleUpSellProductsTest (reported by @m2-assistant[bot]) has been fixed in magento/magento2#34110 by @kate-kyzyma in 2.4-develop branch
   Related commits:
     1. 1d178a9e4cf8559ee74bd0033c5b02595712f87c
     2. efe3d45b21dbe0f9b822fddf1c63e191786232a1

 - magento/magento2#34313: [Issue] Replace repetitive actions with Action Groups in AdminEditRelatedBundleProductTest, EnableDisableBundleProductStatusTest, AdminMassDeleteBundleProductsTest (reported by @m2-assistant[bot]) has been fixed in magento/magento2#34099 by @kate-kyzyma in 2.4-develop branch
   Related commits:
     1. a38ed7e586a5fedccc498def60aae49fbd436f81
     2. d063a3478928cd4fa0e2327490193e68cc4576a3
     3. 220b5564f0e4b2ca9a951d21a8b4e1b56eadbcd6
     4. a6d28bcf6ae502b5cc7e85c22e4553be5272a055
     5. 478f7655eefb47481e584565624a0c27baa5efd5
     6. be43f10c198b4f9d108b89dbf12d6a49931aa9ab
     7. fc4ea7b3fd057532eaf321dcff29a2f8bd7dddb5

 - magento/magento2#34311: [Issue] Replace repetitive actions with Action Groups in StorefrontConfigurableProductGridViewTest and StorefrontConfigurableProductListViewTest (reported by @m2-assistant[bot]) has been fixed in magento/magento2#34051 by @kate-kyzyma in 2.4-develop branch
   Related commits:
     1. 1c355743de1052d9cdbb989669be8edf0e8284f0
     2. 2742d8344156811c89c3143cd95ce7a82526b2a3

 - magento/magento2#34312: [Issue] Replace repetitive actions with Action Groups in AdminAddBundleItemsTest (reported by @m2-assistant[bot]) has been fixed in magento/magento2#34075 by @kate-kyzyma in 2.4-develop branch
   Related commits:
     1. 36ad6b75064d924cdcc7efd3533a10528e1f4f5c
     2. 76a5004953c4335a47f0ed843011c8818adeb893
     3. 1d0748352926e3506eeb2aeae8bdea838bf9ddcd
     4. a4d9bc1a6fb36b45ffb3f347e7a73574a240b348
     5. 61c8530d896178f5a48b2e767115514807b209cb

 - magento/magento2#33531: [Issue] Removed CliCacheFlushActionGroup usage for CatalogSearch and other modules (reported by @m2-assistant[bot]) has been fixed in magento/magento2#33530 by @sergiy-v in 2.4-develop branch
   Related commits:
     1. fafdb234186d92b2e00df4ad562b85fc91d091d7

sidolov committed:
ENGCOM-9323: Replace repetitive actions with Action Groups in AdminSimpleProductSetEditContentTest,AdminCreateAndEditSimpleProductSettingsTest, AdminCreateAndEditVirtualProductSettingsTest and AdminConfigurableProductBulkUpdateTest #34302

sidolov committed:
ENGCOM-9322: Replace repetitive actions with Action Groups in AdminAddInStockProductToTheCartTest  #34298

sidolov committed:
ENGCOM-9321: Replace repetitive actions with Action Groups in AdminConfigurableProductAddConfigurationTest, AdminConfigurableProductDisableAnOptionTest, AdminConfigurableProductRemoveAnOptionTest, AdminConfigurableProductRemoveConfigurationTest #34400

sidolov committed:
ENGCOM-9320: Replace repetitive actions with Action Groups in AdminCreateInactiveInMenuFlatCategoryTest #34418

sidolov committed:
ENGCOM-9317: Replace repetitive actions with Action Groups in AdminCreateDuplicateCategoryTest and AdminCreateDuplicateProductTest #34410

sidolov committed:
ENGCOM-9310: Guard Magento_Catalog JavaScript from lack of minicart #34185

sidolov committed:
ENGCOM-9282: Replace repetitive actions with Action Groups in CreateProductAttributeEntityWithReservedKeysTest #34419

sidolov committed:
ENGCOM-9188: Fix accessibility errors on cart and checkout pages #33851

amolina-adobe committed:
Merge remote-tracking branch 'commerce/2.4-develop' into graphql-doc-changes

NadiyaS committed:
B2B-2069: Images from remote directory is throwing errors on lib/internal/Magento/Framework/Filesystem/Directory/Write.php

NadiyaS committed:
Merge branch '2.4-develop' of github.com:magento-arcticfoxes/magento2ce into B2B-2049

xmav committed:
Merge branch '2.4-develop' into platform-health

adifucan committed:
Merge remote-tracking branch 'origin/2.4-develop' into MCP-773

AnnaAPak committed:
Merge branch '2.4-develop' into ref-StorefrontVerifyOrderShipmentForDecimalQuantityTest

Commit:
fix after a CR

magento-engcom-team committed:
ENGCOM-9324: Replace repetitive actions with Action Groups in AdminUpdateFlatCategoryNameAndDescriptionTest #34492

 - Merge Pull Request magento/magento2#34492 from kate-kyzyma/magento2:Refactoring-AdminUpdateFlatCategoryNameAndDescriptionTest
 - Merged commits:
   1. 14b41be898da6fe48ad91a4b523c7ddaaf435d54

oshmyheliuk committed:
Merge branch '2.4-develop' into B2B-2028

NadiyaS committed:
B2B-2049: [AWS S3] Magento\Customer\Model\Metadata\Form\Image File Filesystem hardcoded

sidolov committed:
:arrows_clockwise: [Magento Community Engineering] Community Contributions - 2.4-develop latest changes

Accepted Community Pull Requests:
 - magento/magento2#34302: Replace repetitive actions with Action Groups in AdminSimpleProductSetEditContentTest,AdminCreateAndEditSimpleProductSettingsTest, AdminCreateAndEditVirtualProductSettingsTest and AdminConfigurableProductBulkUpdateTest (by @kate-kyzyma)
 - magento/magento2#34298: Replace repetitive actions with Action Groups in AdminAddInStockProductToTheCartTest  (by @kate-kyzyma)
 - magento/magento2#34400: Replace repetitive actions with Action Groups in AdminConfigurableProductAddConfigurationTest, AdminConfigurableProductDisableAnOptionTest, AdminConfigurableProductRemoveAnOptionTest, AdminConfigurableProductRemoveConfigurationTest (by @kate-kyzyma)
 - magento/magento2#34418: Replace repetitive actions with Action Groups in AdminCreateInactiveInMenuFlatCategoryTest (by @kate-kyzyma)
 - magento/magento2#34410: Replace repetitive actions with Action Groups in AdminCreateDuplicateCategoryTest and AdminCreateDuplicateProductTest (by @kate-kyzyma)
 - magento/magento2#34185: Guard Magento_Catalog JavaScript from lack of minicart (by @fredden)
 - magento/magento2#34419: Replace repetitive actions with Action Groups in CreateProductAttributeEntityWithReservedKeysTest (by @kate-kyzyma)
 - magento/magento2#33851: Fix accessibility errors on cart and checkout pages (by @vovayatsyuk)


Fixed GitHub Issues:
 - magento/magento2#34435: [Issue] Replace repetitive actions with Action Groups in AdminSimpleProductSetEditContentTest,AdminCreateAndEditSimpleProductSettingsTest, AdminCreateAndEditVirtualProductSettingsTest and AdminConfigurableProductBulkUpdateTest (reported by @m2-assistant[bot]) has been fixed in magento/magento2#34302 by @kate-kyzyma in 2.4-develop branch
   Related commits:
     1. 67c83d9f2b6d4ea44ab8e38711adb92d53b4e6c6
     2. f550f195db5b616eac28d6f4d3f6ff1d01bc6898
     3. 48162ef55dee01da99c998d582e57446cf80e621
     4. 9a7145222a011326feb5de172673fa0b69ea0e3e

 - magento/magento2#34512: [Issue] Replace repetitive actions with Action Groups in AdminAddInStockProductToTheCartTest  (reported by @m2-assistant[bot]) has been fixed in magento/magento2#34298 by @kate-kyzyma in 2.4-develop branch
   Related commits:
     1. 90e3b3a555e4a78b6a001850de9e5a009d47d145
     2. 8710edf8487cc31bf72efe63253f6eef2dc02229
     3. a3479ea072924e6d9a3c6596e6d5deb1cc0d9ca1
     4. a1290c6e0c67695920ac3ee1ff3be86bee0b16c0
     5. f548447222dc01519d351ad01af32e63bc6fe657

 - magento/magento2#34511: [Issue] Replace repetitive actions with Action Groups in AdminConfigurableProductAddConfigurationTest, AdminConfigurableProductDisableAnOptionTest, AdminConfigurableProductRemoveAnOptionTest, AdminConfigurableProductRemoveConfigurationTest (reported by @m2-assistant[bot]) has been fixed in magento/magento2#34400 by @kate-kyzyma in 2.4-develop branch
   Related commits:
     1. 2d2a20a4fd7bae5adb1095248a71d3b33acd068a
     2. 3f45e23a43d807fadbdccc53f9d59dbaf3fda394

 - magento/magento2#34510: [Issue] Replace repetitive actions with Action Groups in AdminCreateInactiveInMenuFlatCategoryTest (reported by @m2-assistant[bot]) has been fixed in magento/magento2#34418 by @kate-kyzyma in 2.4-develop branch
   Related commits:
     1. 94a7cdc71175841cabea52d597fe6d45edea716f

 - magento/magento2#34414: [Issue] Replace repetitive actions with Action Groups in AdminCreateDuplicateCategoryTest and AdminCreateDuplicateProductTest (reported by @m2-assistant[bot]) has been fixed in magento/magento2#34410 by @kate-kyzyma in 2.4-develop branch
   Related commits:
     1. 9972de05b32557998674a528c7b32c3f82dde0b5
     2. c934b510419356b9e369c185bb3b69d055e18897
     3. ef654e5b195d9bb46bbd89b479f3cb17173cc0fc
     4. 1faca0100ff04513e1c306454c02d703890d7af1

 - magento/magento2#34513: [Issue] Guard Magento_Catalog JavaScript from lack of minicart (reported by @m2-assistant[bot]) has been fixed in magento/magento2#34185 by @fredden in 2.4-develop branch
   Related commits:
     1. 209b24bcb9e449123689817512e65d8e9eda32e0
     2. bdfd02ae0d06943a9aa440f3da6e489019ca77b0

 - magento/magento2#34422: [Issue] Replace repetitive actions with Action Groups in CreateProductAttributeEntityWithReservedKeysTest (reported by @m2-assistant[bot]) has been fixed in magento/magento2#34419 by @kate-kyzyma in 2.4-develop branch
   Related commits:
     1. ec0e64904fe0b4795fa3059f89f45c867870d3a5
     2. 2377b2fc4f860a1f9f6c7f0bd0905d9c1ccf679c
     3. 49796bb78d69061ff07a13a235b9813c37e291e9

 - magento/magento2#34483: [Issue] Fix accessibility errors on cart and checkout pages (reported by @m2-assistant[bot]) has been fixed in magento/magento2#33851 by @vovayatsyuk in 2.4-develop branch
   Related commits:
     1. 6c1807be1904232b877dabaadff8e6a810d16b30
     2. a35ee4e05f591c9ab9cf28326ffdaa55c541a01f
     3. a9e5f7b32acb19856b57cada7afbcb2ea9b7c4e5
     4. 8c0e20ca32d067e20f7162c53b94f8a57159ebf4
     5. 9f676cb39ed99b57d683c150c3e896549385eb26
     6. 777e337c5daa58b256fe69bf563d52b39dfda3e3
     7. e0197b9bb9296196db80e1432257a9d1a31e3259

magento-engcom-team committed:
ENGCOM-9325: Replace repetitive actions with Action Groups in AdminCreateInactiveFlatCategoryTest and AdminUpdateFlatCategoryAndAddProductsTest #34470

 - Merge Pull Request magento/magento2#34470 from kate-kyzyma/magento2:Refactoring-AdminCreateInactiveFlatCategoryTest
 - Merged commits:
   1. 8fad53d57eafbe9780512ffb9f4ba09e984788cf
   2. 22b786b99e2befa0a38291875da3a090aa462fa5
   3. c9e03170a49da335beabea02a5bd63a79ebfd627
   4. 64f54573acc319c47d97a126c918d85039ed13e0
   5. 3076e8d5679c276337d3e0f05b0f2df652d881f3
   6. ab51a28871739f48e13f82e4b02f34c5a04ffba8
   7. 5a345aed98167a393dcadb55799ebf10b5fdd527

oshmyheliuk committed:
B2B-2024: [AWS S3] [Integration Tests]: Investigate Test Failures in cms module

oshmyheliuk committed:
Merge remote-tracking branch 'origin/2.4-develop' into B2B-2024

magento-engcom-team committed:
ENGCOM-9326: Replace repetitive actions with Action Groups in AdminCreateCatalogPriceRuleForCustomerGroupTest #34114

 - Merge Pull Request magento/magento2#34114 from kate-kyzyma/magento2:Refactoring-AdminCreateCatalogPriceRuleForCustomerGroupTest
 - Merged commits:
   1. 9a71527318e5643d44f723fc4ce5ee1a569f59bb
   2. f39b060b19f3c3a7e09652fb2b2e6e04a3c013cd
   3. ed288a2c3fdc26b863cebf943342f6ba08ae7945
   4. 87d3fb2e9908264b795903a42a0d0a5a6d5a8125

magento-engcom-team committed:
ENGCOM-9327: [MFTF] StorefrontVerifyOrderShipmentForDecimalQuantityTest refactoring #34152

 - Merge Pull Request magento/magento2#34152 from AnnaAPak/magento2:ref-StorefrontVerifyOrderShipmentForDecimalQuantityTest
 - Merged commits:
   1. 2bdb295da1969e49d100982411d8df21839d2552
   2. a4626122990645178645a555ff8cf1e499057e88

magento-engcom-team committed:
ENGCOM-9328: Update Cms Module README Content #34442

 - Merge Pull Request magento/magento2#34442 from ravi-chandra3197/magento2:Cms_Module
 - Merged commits:
   1. 07cba6ecb215756c711c50e26f42c06e494511ed

cspruiell committed:
Merge branch '2.4-develop' into graphql-doc-changes

magento-engcom-team committed:
ENGCOM-9329: Update Catalog Module README Content #34423

 - Merge Pull Request magento/magento2#34423 from ravi-chandra3197/magento2:Catalog_Module
 - Merged commits:
   1. dc0289cf2fd49ea9e8adfcddd5289136e9d3f3b7

oshmyheliuk committed:
B2B-2024: [AWS S3] [Integration Tests]: Investigate Test Failures in cms module

sidolov committed:
ENGCOM-9329: Update Catalog Module README Content #34423

sidolov committed:
ENGCOM-9328: Update Cms Module README Content #34442

sidolov committed:
ENGCOM-9327: [MFTF] StorefrontVerifyOrderShipmentForDecimalQuantityTest refactoring #34152

sidolov committed:
ENGCOM-9326: Replace repetitive actions with Action Groups in AdminCreateCatalogPriceRuleForCustomerGroupTest #34114

sidolov committed:
ENGCOM-9325: Replace repetitive actions with Action Groups in AdminCreateInactiveFlatCategoryTest and AdminUpdateFlatCategoryAndAddProductsTest #34470

sidolov committed:
ENGCOM-9324: Replace repetitive actions with Action Groups in AdminUpdateFlatCategoryNameAndDescriptionTest #34492

sidolov committed:
ENGCOM-9254: Resolve height modal issue on iOS #34290

sidolov committed:
ENGCOM-9207: 33465 Varnish cache adding top menu category ids to all page cache tags #33468

sidolov committed:
ENGCOM-9181: [MFTF] Adding AdminCheckOrderStatusInGridActionGroup #33727

sidolov committed:
ENGCOM-9046: MFTF: Reset Admin Password With Enabled Captcha #32877

NadiyaS committed:
B2B-2069: Images from remote directory is throwing errors on lib/internal/Magento/Framework/Filesystem/Directory/Write.php

NadiyaS committed:
Merge branch '2.4-develop' of github.com:magento-arcticfoxes/magento2ce into B2B-2069

Commit:
Merge branch '2.4-develop' into Refactoring-AdminCreateInactiveFlatCategoryTest

karyna-t committed:
Added fixes for static tests found on platform health

avattam06 committed:
Merge branch '2.4-develop' into B2B-2028

xmav committed:
Merge branch '2.4-develop' into platform-health

magento-mts-svc committed:
merge magento-commerce/2.4-develop into magento-performance/MCP-773

NadiyaS committed:
B2B-2031: [AWS S3] [Integration Tests]: Investigate Test Failures in MediaGallery module

sidolov committed:
Update ProcessLayoutRenderElement.php

adifucan committed:
Merge remote-tracking branch 'performance-ce/MCP-771' into MCP-773

NadiyaS committed:
B2B-2031: [AWS S3] [Integration Tests]: Investigate Test Failures in MediaGallery module

NadiyaS committed:
B2B-2031: [AWS S3] [Integration Tests]: Investigate Test Failures in MediaGallery module

slopukhov committed:
Merge branch '2.4-develop' of github.com:magento-commerce/magento2ce into MCP-770

ArtemVasyliev committed:
Merge branch 'magento-commerce:2.4-develop' into L3-PR-20211001

oshmyheliuk committed:
B2B-2024: [AWS S3] [Integration Tests]: Investigate Test Failures in cms module

oshmyheliuk committed:
B2B-2024: [AWS S3] [Integration Tests]: Investigate Test Failures in cms module

joanhe committed:
Merge pull request #7167 from magento-arcticfoxes/B2B-2028

B2B-2028: [AWS S3] [Integration Tests]: Investigate Test Failures in CustomerCustomAttributes module

vzabaznov committed:
Merge branch '2.4-develop' into MCP-773

xmav committed:
Merge branch '2.4-develop' into platform-health

vzabaznov committed:
Merge pull request #7182 from magento-performance/MCP-773

[performance] MCP-773: [GraphQL] Error in applying a coupon during a checkout

xmav committed:
Merge branch '2.4-develop' into platform-health

ArtemVasyliev committed:
Merge branch 'magento-commerce:2.4-develop' into L3-PR-20211001

slopukhov committed:
Merge branch '2.4-develop' of github.com:magento-commerce/magento2ce into MCP-770

NadiyaS committed:
B2B-2031: [AWS S3] [Integration Tests]: Investigate Test Failures in MediaGallery module

NadiyaS committed:
Merge branch '2.4-develop' of github.com:magento-arcticfoxes/magento2ce into B2B-2031

NadiyaS committed:
Merge branch 'B2B-2049' into B2B-2069

NadiyaS committed:
Merge branch '2.4-develop' of github.com:magento-arcticfoxes/magento2ce into B2B-2069

oshmyheliuk committed:
Merge remote-tracking branch 'origin/2.4-develop' into B2B-2024

loginesta committed:
AC-1509: Remove outdated JSCS static tests from magento

amolina-adobe committed:
Merge remote-tracking branch 'commerce/2.4-develop' into graphql-doc-changes

victor-v-rad committed:
Merge branch '2.4-develop' into ACP2E-10

thiaramus committed:
ACP2E-31: Free Shipping does not show in totals on invoice page

magento-engcom-team committed:
ENGCOM-9325: Replace repetitive actions with Action Groups in AdminCreateInactiveFlatCategoryTest and AdminUpdateFlatCategoryAndAddProductsTest #34470

 - Merge Pull Request magento/magento2#34470 from kate-kyzyma/magento2:Refactoring-AdminCreateInactiveFlatCategoryTest
 - Merged commits:
   1. 8fad53d57eafbe9780512ffb9f4ba09e984788cf
   2. 22b786b99e2befa0a38291875da3a090aa462fa5
   3. c9e03170a49da335beabea02a5bd63a79ebfd627
   4. 64f54573acc319c47d97a126c918d85039ed13e0
   5. 3076e8d5679c276337d3e0f05b0f2df652d881f3
   6. ab51a28871739f48e13f82e4b02f34c5a04ffba8
   7. 5a345aed98167a393dcadb55799ebf10b5fdd527
   8. 3d3f70b5c286efb8c347bc6f1f3721cbbb80c618

Commit:
Merge remote-tracking branch 'origin/ENGCOM-9325-magento-magento2-34470' into 2.4-develop-prs

thiaramus committed:
MC-43163: [Cloud] [Premier Support] Cannot update only extension attributes via REST (POST: /rest/V1/orders)

thiaramus committed:
MC-43163: [Cloud] [Premier Support] Cannot update only extension attributes via REST (POST: /rest/V1/orders)

o-dubovyk committed:
ACP2E-212: Customer Grid loading on the Admin Order creation page is slow

- fix static fail

oshmyheliuk committed:
B2B-2024: [AWS S3] [Integration Tests]: Investigate Test Failures in cms module

ArtemVasyliev committed:
Merge branch 'MC-43163' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

ArtemVasyliev committed:
Merge branch 'ACP2E-31' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

ArtemVasyliev committed:
Merge branch 'ACP2E-212' of https://github.com/magento-l3/magento2ce into L3-PR-20211001

cpartica committed:
Merge pull request #7152 from magento-honey-badgers/graphql-doc-changes

[honey] DOC-472: Update GraphQL annotations

ArtemVasyliev committed:
Merge branch 'magento-commerce:2.4-develop' into L3-PR-20211001

victor-v-rad committed:
Merge branch '2.4-develop' into ACP2E-10

NadiyaS committed:
Merge branch 'B2B-2031' into B2B-2069

oshmyheliuk committed:
B2B-2026: [AWS S3] [Integration Tests]: Investigate Test Failures in ConfigurableImportExport module

dhorytskyi committed:
Merge pull request #7171 from magento-l3/ACP2E-10

ACP2E-10: Incorrect Discount: Two Cart rules with and without coupon

ArtemVasyliev committed:
Merge branch 'magento-commerce:2.4-develop' into L3-PR-20211001

xmav committed:
Merge remote-tracking branch 'karyna/php8.1-compatibility/fix-unit-tests-core' into platform-health

xmav committed:
Merge branch 'platform-health' of github.com:magento-commerce/magento2ce into platform-health

xmav committed:
Merge remote-tracking branch 'commerce/2.4-develop' into platform-health

xmav committed:
Merge pull request #7197 from magento-commerce/platform-health

[Platform Health] Dependency update and fixes

loginesta committed:
Merge branch '2.4-develop' of github.com:magento-commerce/magento2ce into AC-1509

hostep committed:
Upgraded phpstan to 1.x

ArtemVasyliev committed:
Merge branch 'magento-commerce:2.4-develop' into L3-PR-20211001

joanhe committed:
Merge remote-tracking branch 'upstream/2.4-develop' into B2B-2069

dhorytskyi committed:
Merge pull request #7188 from magento-l3/L3-PR-20211001

L3 pr 20211001

slopukhov committed:
Merge branch '2.4-develop' of github.com:magento-commerce/magento2ce into MCP-770

NadiyaS committed:
Merge branch '2.4-develop' of github.com:magento-arcticfoxes/magento2ce into B2B-2069

NadiyaS committed:
Merge remote-tracking branch 'origin/B2B-2026' into B2B-2069

vzabaznov committed:
Merge pull request #7189 from magento-performance/MCP-770

[performance] MCP-770: Add performance measurement of checkout with a large number of products in the cart for regular PAT build

NadiyaS committed:
Merge branch '2.4-develop' of github.com:magento-commerce/magento2ce into B2B-2069

oshmyheliuk committed:
B2B-2020: [AWS S3] [Integration Tests]: Investigate Test Failures in AdvancedPricingImportExport module

loginesta committed:
Merge branch '2.4-develop' of github.com:magento-commerce/magento2ce into AC-1509

NadiyaS committed:
B2B-2069: Images from remote directory is throwing errors on lib/internal/Magento/Framework/Filesystem/Directory/Write.php
 - changes to ignore phpcs checks

oshmyheliuk committed:
Merge branch '2.4-develop' into B2B-2024

NadiyaS committed:
B2B-2069: Images from remote directory is throwing errors on lib/internal/Magento/Framework/Filesystem/Directory/Write.php
 - suppress warning for data provider

sivaschenko committed:
Merge pull request #7205 from magento-lynx/AC-1509

[Lynx] Remove outdated JSCS static tests from magento

NadiyaS committed:
Merge branch '2.4-develop' of github.com:magento-commerce/magento2ce into B2B-2069

andrewbess committed:
Fixed constraint for `phpstan/phpstan` according to reviewer request

karyna-t committed:
fixes for integration tests errors on PHP 8.1

andrewbess committed:
Merge branch '2.4-develop' into upgrade-phpstan-to-first-stable-version

joanhe committed:
Merge pull request #7196 from magento-arcticfoxes/B2B-2069

B2B-2069: Images from remote directory is throwing errors on lib/internal/Magento/Framework/Filesystem/Directory/Write.php

xmav committed:
Merge remote-tracking branch 'trigger/imported-hostep-magento2-34555' into ph-delivery

xmav committed:
Merge remote-tracking branch 'trigger/imported-SilinMykola-magento2-34446' into ph-delivery

xmav committed:
Merge pull request #7213 from magento-trigger/ph-delivery

[Platform Health] Updates for PHP8.1

Commit:
fix after a CR

oshmyheliuk committed:
B2B-2020: [AWS S3] [Integration Tests]: Investigate Test Failures in AdvancedPricingImportExport module

oshmyheliuk committed:
Merge remote-tracking branch 'origin/2.4-develop' into B2B-2020

sidolov committed:
Merge branch '2.4-develop' into 2.4-develop-prs

joanhe committed:
Merge remote-tracking branch 'upstream/2.4-develop' into B2B-2024

oshmyheliuk committed:
B2B-2020: [AWS S3] [Integration Tests]: Investigate Test Failures in AdvancedPricingImportExport module

xmav committed:
Merge remote-tracking branch 'trigger/imported-SilinMykola-magento2-34450' into ph-delivery

xmav committed:
Merge remote-tracking branch 'trigger/imported-karyna-tsymbal-atwix-magento2-34617' into ph-delivery

oshmyheliuk committed:
Merge branch 'B2B-2020' into B2B-2024

joanhe committed:
Merge pull request #7206 from magento-engcom/2.4-develop-prs

[Magento Community Engineering] Community Contributions - 2.4-develop

joanhe committed:
Merge remote-tracking branch 'upstream/2.4-develop' into B2B-2024

xmav committed:
Merge branch '2.4-develop' into ph-delivery

joanhe committed:
Merge pull request #7193 from magento-arcticfoxes/B2B-2024

B2B-2024: [AWS S3] [Integration Tests]: Investigate Test Failures in cms module

xmav committed:
Merge branch '2.4-develop' into ph-delivery

xmav committed:
Merge pull request #7215 from magento-trigger/ph-delivery

[Platform Health] Updates for PHP8.1

vshcheglov commented:
Added a pull request with additional changes based on 2.4-develop branch being merged to it.
https://github.com/magento/magento2/pull/34856
